kolibri/core/assets/src/views/buttons-and-links/k-button.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+
+  <button
+    ref="button"
+    :class="buttonClasses"
+    :type="type"
+    :disabled="disabled"
+    @click="handleClick">
+    {{ text }}
+  </button>
+
+</template>
+
+
+<script>
+
+  import { validator } from './appearances.js';
+  import buttonClassesMixin from './buttonClassesMixin.js';
+
+  /**
+   * The kButton component is used to trigger actions
+   */
+  export default {
+    name: 'kButton',
+    mixins: [buttonClassesMixin],
+    props: {
+      /**
+       * Button label text
+       */
+      text: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Whether or not button is disabled
+       */
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * HTML button 'type' attribute (e.g. 'submit', 'reset')
+       */
+      type: {
+        type: String,
+        default: 'button',
+      },
+      /**
+       * Button appearance: 'raised-button', 'flat-button', or 'basic-link'
+       */
+      appearance: {
```

Comment:
Kind of minor, but can you move the appearance and primary props above disabled and type.